### PR TITLE
Add dyz60

### PR DESCRIPTION
**Pull request by dayatz**
# dyz60

60% layout has flexcut with multiple layout and mounting support
pcb

* Keyboard Maintainer: [dayatz](https://github.com/dayatz)
* Hardware Supported: dyz60, ATMega32u4
* Hardware Availability: Local Indonesia GB.

Make example for this keyboard (after setting up your build environment):

    make dyz/dyz60:default

Flashing example for this keyboard:

    make dyz/dyz60:default:flash

See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).

### CODE REVIEW

File: keyboards/dyz/dyz60/config.h [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xD772
+#define PRODUCT_ID      0x000A
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    dyz
+#define PRODUCT         dyz60
+#define DESCRIPTION     60% with flexcut and multiple mounts support
```
Comment (drashna) [inline keyboards/dyz/dyz60/config.h:28]:
```suggestion
```

File: keyboards/dyz/dyz60/config.h [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+*/
+#define MATRIX_ROW_PINS { B0, E6, F0, F5, F4 }
+#define MATRIX_COL_PINS { F6, F7, B3, B2, B1, D2, C7, C6, B6, B5, B4, D7, D6, D4 }
+#define UNUSED_PINS
+
+/* COL2ROW, ROW2COL*/
+#define DIODE_DIRECTION COL2ROW
+
+#define RGB_DI_PIN B7
+#define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/dyz/dyz60/config.h:52]: could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/dyz/dyz60/config.h [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+#define TAP_CODE_DELAY 10 // Need this delay for some keys like VOLD/VOLU to work
+#define ENCODERS_PAD_A { D0 }
+#define ENCODERS_PAD_B { D1 }
+
+// python3 util/vial_generate_keyboard_uid.py
+#define VIAL_KEYBOARD_UID {0xBF, 0x0D, 0x8D, 0xDF, 0xDB, 0x11, 0x0C, 0xA5}
+// encoder per layers
+#define VIAL_ENCODER_DEFAULT { KC_VOLD, KC_VOLU, KC_LEFT, KC_RIGHT, KC_TRNS, KC_TRNS }
+#define VIAL_ENCODER_KEYCODE_DELAY 10
```
Comment (drashna) [inline keyboards/dyz/dyz60/config.h:74]: Please remove these.  These are VIAL specific settings, and are not used in this repo. 
```suggestion
```

File: keyboards/dyz/dyz60/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+#define _______ KC_TRNS
```
Comment (drashna) [inline keyboards/dyz/dyz60/keymaps/default/keymap.c:17]: These defines aren't necessary, as these are actually defined in the core.  You can safely remove them. 

https://github.com/qmk/qmk_firmware/blob/a0309db983150fd0197eb620a6ba552d90d29f93/quantum/quantum_keycodes.h#L30-L31


```suggestion
```

File: keyboards/dyz/dyz60/keymaps/vial/keymap.c [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+#define _______ KC_TRNS
```
Comment (drashna) [inline keyboards/dyz/dyz60/keymaps/vial/keymap.c:17]: These defines aren't necessary, as these are actually defined in the core.  You can safely remove them. 

https://github.com/qmk/qmk_firmware/blob/a0309db983150fd0197eb620a6ba552d90d29f93/quantum/quantum_keycodes.h#L30-L31


```suggestion
```

File: keyboards/dyz/dyz60/keymaps/vial/rules.mk [hunk 6]
```diff
@@ -0,0 +1,8 @@
+VIA_ENABLE = yes
```
Comment (drashna) [inline keyboards/dyz/dyz60/keymaps/vial/rules.mk:1]: Please remove.

File: keyboards/dyz/dyz60/keymaps/vial/vial.json [hunk 7]
```diff
@@ -0,0 +1,211 @@
+{
```
Comment (drashna) [inline keyboards/dyz/dyz60/keymaps/vial/vial.json:1]: Please remove.

File: keyboards/dyz/dyz60/rules.mk [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (drashna) [inline keyboards/dyz/dyz60/rules.mk:11]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/dyz/dyz60/keymaps/via/keymap.c [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+ */
+#include QMK_KEYBOARD_H
+
+enum layer_names {
+    _BASE,
+    _FN,
+    _FN2
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
```
Comment (drashna) [inline keyboards/dyz/dyz60/keymaps/via/keymap.c:24]: This needs to have 4 layers, or it needs to have the define to limit it to 3.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/dyz/dyz60/keymaps/via/keymap.c (hunk 9 above)
Comment (dayatz) [reply keyboards/dyz/dyz60/keymaps/via/keymap.c:24]: ok, I'll limit the layers to 3

**Review by drashna — APPROVED** [review]

**Comment by dayatz** [comment]
any update on this MR ?

File: keyboards/dyz/dyz60/rules.mk [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+MOUSEKEY_ENABLE = no       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+ENCODER_ENABLE = yes
```
Comment (fauxpark) [inline keyboards/dyz/dyz60/rules.mk:20]:
```suggestion
BOOTMAGIC_ENABLE = lite     # Enable Bootmagic Lite
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
ENCODER_ENABLE = yes
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by fauxpark — CHANGES_REQUESTED** [review]
Whoops, missing an info.json.

**Comment by dayatz** [comment]
please kindly check my info.json

File: keyboards/dyz/dyz60/info.json [hunk 11]
```diff
@@ -0,0 +1,32 @@
+{
+    "keyboard_name": "dyz60",
+    "url": "https://github.com/dayatz/mechanical-keyboards/tree/master/dyz60",
+    "maintainer": "dayatz",
+    "width": 15,
+    "height": 5,
```
Comment (fauxpark) [inline keyboards/dyz/dyz60/info.json:6]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by dayatz** [comment]
if there are any changes needed, lemme know...

File: keyboards/dyz/dyz60/rules.mk [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
```
Comment (fauxpark) [inline keyboards/dyz/dyz60/rules.mk:21]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/dyz/dyz60/rules.mk (hunk 12 above)
Comment (dayatz) [reply keyboards/dyz/dyz60/rules.mk:21]: should I also remove `AUDIO_ENABLE = no` ?

Comment (fauxpark) [reply keyboards/dyz/dyz60/rules.mk:21]: No.

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**